cgi-bin/LJ/Lang.pm

```diff
@@ -612,7 +612,7 @@ sub get_text {
         ? $from_files->()
         : $from_db->();
 
-    if ($vars) {
+    if ( defined $vars && %$vars ) {
```

Comment:
nit: This still has some edge cases (defined is true for arrayrefs for example but then will fail when it's coerced into a hash?) so the typically way of doing this is,

```perl
if ( ref $vars eq 'HASH' && %$vars ) {
```